O2/ML/compute_efficiency_df.py

```diff
@@ -0,0 +1,193 @@
+    Outputs
+    - effifiency: 3D array efficiency[pT index][class][efficiency value]
+    """
+    
+    pt_mins = pt_bins[:-1]
+    pt_maxs = pt_bins[1:]
+
+    efficiency = []
+    for ipt, (pt_min, pt_max) in enumerate(zip(pt_mins, pt_maxs)):
+        efficiency.append([])
+        df_pt = df.query(f"{pt_min} < {pt_prong} < {pt_max}")
+        den = len(df_pt)
+        for iclas, clas in enumerate(["Bkg", "Prompt", "Nonprompt"]):
+            operand = "<" if clas == "Bkg" else ">"
+            efficiency[ipt].append([])
+            for thr in thresholds:
+                if iclas == 0:
+                    df_cut = df_pt.query(f"ML_output_{clas} < {thr}")
+                else:
+                    df_cut = df_pt.query(f"ML_output_Bkg < {BDT_cuts[0]} and ML_output_{clas} > {thr}")
+                efficiency[ipt][iclas].append(np.float(len(df_cut)) / den)
+
+    return efficiency
+
+def plot_efficiency_vs_BDTscore(eff, channel, pt_bins, BDT_cuts, thresholds):
+    plt.rcParams['axes.grid'] = True
+    plt.figure(f"{channel} BDT efficiencies", figsize=(16, 5))
+    # choice of pt value
+    ipt = 0
+    for label in range(3):
+        for iclas, clas in enumerate(["Bkg", "Prompt", "Nonprompt"]):
+            plt.subplot(1,3,iclas+1)
+            plt.scatter(thresholds, eff[label][0][iclas])
+            plt.ylim(0.00001,1.)
+            plt.yscale('log')
+            plt.xlabel("BDT score")
+            if iclas == 0:
+                plt.ylabel(f"Efficiency for {clas} at pT = {pt_bins[ipt]} GeV/c")
+            else:
+                plt.ylabel(f"Efficiency for {clas} at pT = {pt_bins[ipt]} GeV/c (Bkg BDT < {BDT_cuts[0]})")
+        plt.legend(['Bkg', 'Prompt', 'Nonprompt'])
+
+    plt.show()
+
+def plot_efficiency_vs_pt(eff_vs_pt, channel, pt_bins, BDT_cuts):
+    plt.rcParams['axes.grid'] = True
+    plt.figure(f"{channel} BDT efficiencies", figsize=(16, 5))
+    # configure binning for the plot
+    bins_width, bins_mean = config_bins(pt_bins)
+
+    for label in range(3):
+        for iclas, clas in enumerate(["Bkg", "Prompt", "Nonprompt"]):
+            plt.subplot(1,3,iclas+1)
+            plt.errorbar(bins_mean, eff_vs_pt[label][iclas], xerr = bins_width, fmt = 'o', markersize=5, elinewidth = 2, capsize=4)
+            plt.ylim(0.001,1.)
+            plt.yscale('log')
+            plt.xlabel(r"$p_T$ (GeV/c)")
+            if iclas == 0:
+                plt.ylabel(f"Efficiency for {clas} BDT cut at {BDT_cuts[iclas]}")
+            else:
+                plt.ylabel(f"Efficiency for {clas} BDT cut at {BDT_cuts[iclas]} (Bkg BDT < {BDT_cuts[0]})")
+        plt.legend(['Bkg', 'Prompt', 'Nonprompt'])
+
+    plt.show()
+
+def main(config):
+    """
+    Main function
+    """
+
+    infile = config["input_file"]
+    channel = config["channel"]
+    pt_bins = config["pt_bins"]
+    bkg_BDT_cut = config["BDT_cut"]["Bkg"]
+    prompt_BDT_cut = config["BDT_cut"]["Prompt"]
+    nonprompt_BDT_cut = config["BDT_cut"]["Nonprompt"]
```

Comment:
Here I would simply get the `BDT_cut` from the config (which is already a dictionary) instead of defining later a list
```suggestion
    BDT_cuts = config["BDT_cut"]
```